crates/pathfinder/src/p2p_network/sync_handlers.rs

```diff
@@ -79,11 +79,9 @@ fn fetch_block_headers(
             transaction_count: transaction_count
                 .try_into()
                 .context("Too many transactions")?,
-            // TODO: how to get these values. We'd have to store these (along with the parent hash) as part of the starknet_blocks table
-            // so that we can implement this effectively. Unfortunately re-computing these values is slow...
-            transaction_commitment: Felt::ZERO,
+            transaction_commitment: block.transaction_commitment.unwrap_or_default().0,
             event_count: 0,
-            event_commitment: Felt::ZERO,
+            event_commitment: block.event_commitment.unwrap_or_default().0,
```

Comment:
One alternative is calculating it on the fly and then storing it as well. However, that won't scale for 1000's of headers per p2p request..

Reply:
Yes, I don't think we can default this to zero either...

I _think_ what we could do is doing both of these:

- computing the commitment on-the-fly, which won't scale
- release the next "official" snapshot with commitments in place for historical blocks -- if you're using that then you'll be getting reasonable P2P performance

Reply:
We're also allowed to make the assumption that p2p users will have new snapshots / fresh databases.

Which means we could just fail this instead of doing the extra effort?

Reply:
Well, if only there was a mean of returning an error in the P2P protocol. We'll have to talk about this next week... But yes, that's an option too. 

Ideally we should be able to check if the database is OK on startup when running with P2P enabled?

Reply:
Makes sense, now returning error in case any commitment is missing from the received header.